include/revng/ABIAnalyses/Generated/DeadRegisterArgumentsOfFunction.h

```diff
@@ -0,0 +1,169 @@
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Value.h"
+
+#include "revng/ABIAnalyses/Common.h"
+#include "revng/BasicAnalyses/GeneratedCodeBasicInfo.h"
+#include "revng/MFP/MFP.h"
+#include "revng/Model/Binary.h"
+#include "revng/StackAnalysis/StackAnalysis.h"
+#include "revng/Support/revng.h"
+
+namespace ABIAnalyses::DeadRegisterArgumentsOfFunction {
+
+using namespace ABIAnalyses;
+using Register = model::Register::Values;
+using State = model::RegisterState::Values;
+
+struct CoreLattice {
+
+  // using LatticeElement = model::RegisterState::Values;
+
+  enum LatticeElement { Maybe, NoOrDead, Unknown };
+
+  static const LatticeElement DefaultLatticeElement = Maybe;
+
+  using TransferFunction = ABIAnalyses::TransferKind;
+
+  static bool
+  isLessOrEqual(const LatticeElement &Lh, const LatticeElement &Rh) {
+    return Lh == Rh
+           || (Lh == LatticeElement::Maybe && Rh == LatticeElement::Unknown)
+           || (Lh == LatticeElement::NoOrDead && Rh == LatticeElement::Maybe)
+           || (Lh == LatticeElement::NoOrDead && Rh == LatticeElement::Unknown);
+  }
+
+  static LatticeElement
+  combineValues(const LatticeElement &Lh, const LatticeElement &Rh) {
+    if ((Lh == LatticeElement::Maybe && Rh == LatticeElement::NoOrDead)
+        || (Lh == LatticeElement::NoOrDead && Rh == LatticeElement::Maybe)) {
+      return LatticeElement::Maybe;
+    } else if ((Lh == LatticeElement::Maybe && Rh == LatticeElement::Unknown)
+               || (Lh == LatticeElement::NoOrDead
+                   && Rh == LatticeElement::Unknown)
+               || (Lh == LatticeElement::Unknown && Rh == LatticeElement::Maybe)
+               || (Lh == LatticeElement::Unknown
+                   && Rh == LatticeElement::NoOrDead)) {
+      return LatticeElement::Unknown;
+    }
+    return Lh;
+  }
+
+  static LatticeElement transfer(TransferFunction T, const LatticeElement &E) {
+    switch (T) {
+    case TransferFunction::Read:
+      switch (E) {
+      case LatticeElement::Maybe:
+        return LatticeElement::Unknown;
+      case LatticeElement::Unknown:
+        return LatticeElement::Unknown;
+      case LatticeElement::NoOrDead:
+        return LatticeElement::NoOrDead;
+      default:
+        return E;
+      }
+      return E;
+
+    case TransferFunction::WeakWrite:
+      switch (E) {
+      case LatticeElement::Maybe:
+        return LatticeElement::NoOrDead;
+      case LatticeElement::Unknown:
+        return LatticeElement::Unknown;
+      case LatticeElement::NoOrDead:
+        return LatticeElement::NoOrDead;
+      default:
+        return E;
+      }
+      return E;
+
+    case TransferFunction::Write:
+      switch (E) {
+      case LatticeElement::Maybe:
+        return LatticeElement::NoOrDead;
+      case LatticeElement::Unknown:
+        return LatticeElement::Unknown;
+      case LatticeElement::NoOrDead:
+        return LatticeElement::NoOrDead;
+      default:
+        return E;
```

Comment:
Shouldn't we abort?

Reply:
wouldn't have caught this if I didn't commit the generated code 😆

Reply:
added an abort in case we get a `LatticeElement` that is not valid

I kept the default to identity is for unknown transfer functions (for example if we want to add more kinds of transfer functions that are not useful for a particular analysis)